src/agent.ts

```diff
@@ -0,0 +1,89 @@
+import {
+  Finding,
+  HandleBlock,
+  getEthersProvider,
+  BlockEvent,
+  getAlerts,
+  GetAlerts,
+} from "forta-agent";
+import {
+  DAI_ADDRESS,
+  L1_ESCROW_ARBITRUM,
+  L1_ESCROW_FUNCTION_SIGNATURE,
+  L1_ESCROW_OPTIMISM,
+  BOT_ID,
+  previousBalances,
+} from "./constants";
+import { ethers, Contract, BigNumber } from "ethers";
+import { getL1Finding, checkBlock } from "./utils";
+let chainId: number;
+let prevBal: previousBalances = {
+  previousBlockNumber: 0,
+  prevArbitrumBalance: BigNumber.from(0),
+  prevOptimismBalance: BigNumber.from(0),
+};
+
+export function provideInitialize(provider: ethers.providers.Provider) {
+  return async function initialize() {
+    const networkInfo = await provider.getNetwork();
+    chainId = networkInfo.chainId;
+  };
+}
+
+export function provideHandleBlock(
+  provider: ethers.providers.Provider,
+  getAlerts: GetAlerts,
+): HandleBlock {
+  return async function handleBlock(block: BlockEvent): Promise<Finding[]> {
+    const findings: Finding[] = [];
+
+    try {
+      if (chainId == 1) {
+        const daiContract = new Contract(
+          DAI_ADDRESS,
+          L1_ESCROW_FUNCTION_SIGNATURE,
+          provider,
+        );
+        const result = await getL1Finding(
+          daiContract,
+          block.blockNumber,
+          L1_ESCROW_ARBITRUM,
+          L1_ESCROW_OPTIMISM,
+          prevBal,
+        );
+
+        if (Array.isArray(result)) {
+          findings.push(...result);
+        } else if (result instanceof Finding) {
+          findings.push(result);
+        }
```

Comment:
The `result` can't ever be an array (of Findings) in this case, I think it's better to either modify the return type of `getL1Finding` to be `Promise<Finding | undefined>` (and return `undefined` inside the function when there are no findings, so that you can do `if (result) {...}`, or maybe wrap the Finding object in an array before returning (so that the return type is `Promise<Finding[]`, even if there is one Finding at most) and then just do `findings.push(...result)`